pyproject.toml

```diff
@@ -39,6 +39,8 @@ dependencies = [
     "typing_extensions>=0.0.0",
     "eval_type_backport>=0.0.0",
     "polars>=0.20.22",
+    "GPUtil>=1.4.0",
+    "psutil>=5.9.8",
```

Comment:
Is it worth making these optional? Most people probably won't need them, no?

Reply:
Moved to optional dependencies.